### PR TITLE
Add a demo for file sharing.

### PR DESCRIPTION
**Pull request by jpmedley**
Because this will very soon [be in Chrome beta](https://www.chromestatus.com/feature/4777349178458112).

### CODE REVIEW

**Comment by jpmedley** [comment]
@ericwilligers @mgiuca FYI

**Comment by jpmedley** [comment]
I don't know what I need to do to resolve this issue.

**Comment by ewilligers** [comment]
It is easy to [join WICG](https://www.w3.org/community/wicg/join), you first need a [W3C account](https://www.w3.org/accounts/request).

File: demos/share-files.html [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    </tr>
+  </table>
+  <p><input id="share" type="button" value="Share" />
+     <input id="share-no-gesture" type="button" value="Share without user gesture" /></p>
+  <div id="output"></div>
+  <p>This is a test page for the <a href="https://github.com/WICG/web-share">Web
+  Share API</a>. It will only work in browsers that have implemented the draft
+  proposal. At the time of writing, this works with:</p>
+  <ul>
+    <li>Google Chrome for Android 61 or higher (older versions can enable
```
Comment (ewilligers) [inline demos/share-files.html]: 65 or higher
is needed for file sharing.

Comment (ewilligers) [reply demos/share-files.html]: We can also omit the "Google Chrome for Chrome OS, Windows, or Linux" paragraph. I have temporarily deleted the Web Share Target code from desktop Chrome.

File: demos/share-files.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      }
+
+      try {
+        await navigator.share({files, title, text, url});
+      } catch (error) {
+        logError('Error sharing: ' + error);
+        return;
+      }
+
+      logText('Successfully sent share');
```
Comment (ewilligers) [inline demos/share-files.html]: This line can move inside the try block, before the catch,  Then the return statement in the catch block can be removed.

**Comment by marcoscaceres** [comment]
Marked as non substantive for IPR from ash-nazg.

File: demos/share-files.html [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    async function testWebShareDelay() {
+      await sleep(2000);
```
Comment (ewilligers) [inline demos/share-files.html]: This should be 6000, here and in `share.html`.

The [HTML spec](https://html.spec.whatwg.org/multipage/interaction.html#triggered-by-user-activation) suggests user activation lasts at least 4 seconds. In Chrome it lasts 5 seconds.

File: demos/share-files.html [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      const title_input = document.querySelector('#title');
+      const text_input = document.querySelector('#text');
+      const url_input = document.querySelector('#url');
+      const file_input = document.querySelector('#files');
+
+      const title = title_input.disabled ? undefined : title_input.value;
+      const text = text_input.disabled ? undefined : text_input.value;
+      const url = url_input.disabled ? undefined : url_input.value;
+      const files = file_input.disabled ? undefined : file_input.files;
+      if (!navigator.canShare || !navigator.canShare({files})) {
```
Comment (ewilligers) [inline demos/share-files.html]: The error message needlessly occurs if no files are selected.

Enclose the `if` with
`if (files && files.length > 0) {  }`

File: demos/share-files.html (hunk 1 above)
Comment (jpmedley) [reply demos/share-files.html]: This contains text in a commit I neglected to push before opening the PR. FYI, your 61 is what's in your other demo.

File: demos/share-files.html (hunk 4 above)
Comment (jpmedley) [reply demos/share-files.html]: Done.

File: demos/share-files.html (hunk 2 above)
Comment (jpmedley) [reply demos/share-files.html]: Done. This is also inherited from the other demo.

**Comment by jpmedley** [comment]
I don't know why it didn't recognize me. Not only do I have a W3C account, but I've contributed to this repo within the last week.

**Comment by jpmedley** [comment]
And PTAL.

**Comment by ewilligers** [comment]
> I don't know why it didn't recognize me. Not only do I have a W3C account, but I've contributed to this repo within the last week.

[Joining WICG](https://www.w3.org/community/wicg/join) is a separate step.

**Comment by marcoscaceres** [comment]
Thanks @ericwilligers. Sorry for the process pain @jpmedley - joining the WICG will help. I'm tracking these things also, so can resolve IPR issues quickly.

**Comment by marcoscaceres** [comment]
@jpmedley, I've added you as being part of WICG to the IPR system. Hopefully won't bug you anymore.

**Pull request merged**